consistency-checker/src/main/java/com/redhat/lightblue/migrator/consistency/MigrationJob.java

```diff
@@ -260,38 +268,80 @@ public void run() {
             getJobExecutions().add(currentRun);
 
             configureClients();
+            int jobExecutionPsn = -1;
 
-            saveJobDetails(true);
-
-            Map<String, JsonNode> sourceDocuments = getSourceDocuments();
+            LightblueResponse response = saveJobDetails(jobExecutionPsn);
 
-            Map<String, JsonNode> destinationDocuments = getDestinationDocuments(sourceDocuments);
+            boolean processJob = true;
 
-            List<JsonNode> documentsToOverwrite = getDocumentsToOverwrite(sourceDocuments, destinationDocuments);
+            // verify this job is the first active thread processing the thread.
+            // first extract processed data as MigrationJob
+            JsonNode node = response.getJson().path("processed");
+            try {
+                MigrationJob[] jobs = mapper.readValue(node.traverse(), MigrationJob[].class);
 
+                // should only be one job returned, verify
+                if (jobs.length > 1) {
+                    throw new RuntimeException("Error parsing lightblue response: more than one job returned: " + jobs.length);
+                }
 
-            if (!documentsToOverwrite.isEmpty()) {
-                hasInconsistentDocuments = true;
+                // first incomplete job execution must be for this execution else we don't process it
+                int i = 0;
+                for (MigrationJobExecution execution : jobs[0].getJobExecutions()) {
+                    // if we find an execution that is not our pid but is active
+                    // in the array before ours, we do not get to process
+                    if (!execution.isCompletedFlag() && !pid.equals(execution.getPid())) {
```

Comment:
How are we handling the case where a job has been idle for some period of time? Presumably the process had an error and needs to be attempted again. Maybe the process on the server has been restarted so the pid is no longer the same.